### PR TITLE
#94 fixed event bus related app crushing issue on Android 5.0 Lollipop

### PR DESCRIPTION
**Pull request by bluetoothfx**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
This PR is related to DEV android app. If the application open from android version 5.0 (Lollipop) it crashes immediately & failed to perform application normal activity & shows Error Message "Unfortunately DEV stop working".

What was the error:
I have observed that the project was failing to initiate Event Bus. To properly initiate Event Bus I have followed the guideline
to implement it. Which is available [here](https://greenrobot.org/eventbus/documentation/subscriber-index/).

I need to add annotation processor which was lacking in this project. Which is highly recommended by Greenrobot team.
Which is mentioned [here](https://greenrobot.org/eventbus/documentation/subscriber-index/).
So I have made an application class and put them centrally & found that application working as expected.

## Related Tickets & Documents
Please check following ticket to get details.
#94 

## Screenshots/Recordings (if there are UI changes)
N/A
## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by fdoxyz — CHANGES_REQUESTED** [review]
Thank you for your contribution @bluetoothfx and sorry for the delay in reviewing.

While testing this I saw the crash reported on #98 and also [forem#9809](https://github.com/forem/forem/issues/9809). Apparently when autofill takes control (during login) it calls the `observeNetwork()` method, which in turn calls `EventBus.getDefault().register(this)` and crashes (see screenshot below).

I think this can easily be fixed by calling `EventBus.getDefault().unregister(this)` right before `EventBus.getDefault().register(this)` inside the `observeNetwork()` method from `CustomWebViewClient.kt`. Since this PR refactors the EventBus implementation can you please include this one line for the fix as well? Thanks again for the help.

Screen Shot 2020-09-14 at 13 14 22

**Comment by bluetoothfx** [comment]
@fdoxyz  should I check before unregister or just straight unregister ?
``` 
if(EventBus.getDefault().isRegistered(this)) 
    EventBus.getDefault().unregister(this)
```

**Comment by fdoxyz** [comment]
I think the check would work great 👍

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (bluetoothfx) [reply .idea/jarRepositories.xml:1]: added on .gitignore.

[the remark this answers is not in the document]
Comment (bluetoothfx) [reply .idea/jarRepositories.xml]: Should I remove this .idea/codeStyles/Project.xml also? I have updated kotlin and it appeared.

[the remark this answers is not in the document]
Comment (fdoxyz) [reply .idea/jarRepositories.xml]: I can see that we already had that one in source control. I think for now it's best if we do a separate and thorough PR for working out the files to include/ignore

**Review by fdoxyz — APPROVED** [review]
Tested this and works great, including the suggested bug fix. Thank you again for the contribution @bluetoothfx

**Pull request merged**